src/shell/types.rs

```diff
@@ -124,11 +124,7 @@ impl ShellState {
       }
     } else {
       self.shell_vars.remove(&name);
-      if value.is_empty() {
-        self.env_vars.remove(&name);
-      } else {
-        self.env_vars.insert(name, value.to_string());
-      }
+      self.env_vars.insert(name, value.to_string());
```

Comment:
Thanks! This is more correct. Could you add a test for this in src/shell/test.rs?

Reply:
thanks for the pointer! I grepped with `[test]` so that's way I wasn't able to find that file :) I'll add a test!

Reply:
Added [5690f5a](https://github.com/denoland/deno_task_shell/pull/80/commits/5690f5aba6f3ee35bf8c80baa98eeb7c75152977)